packages/rocketchat-livechat/.app/client/lib/LivechatFileUpload.js

```diff
@@ -0,0 +1,187 @@
+	} else {
+		return callback(file, null);
+	}
+}
+
+function formatBytes(bytes, decimals) {
+	if (bytes === 0) {
+		return '0 Bytes';
+	}
+
+	const k = 1000;
+	const dm = (decimals + 1) || 3;
+
+	const sizes = [
+		'Bytes',
+		'KB',
+		'MB',
+		'GB',
+		'TB',
+		'PB'
+	];
+
+	const i = Math.floor(Math.log(bytes) / Math.log(k));
+
+	return `${ parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) } ${ sizes[i] }`;
+}
+
+function sendFileMessage(file, roomId) {
+	if (!visitor.isSubscribed(roomId)) {	
+		Meteor.call('livechat:startFileUploadRoom', roomId, visitor.getToken(), (error, result) => {
+			if (error) {
+				return showError(error.message);
+			}
+
+			visitor.subscribeToRoom(result.room._id);
+			visitor.setRoom(result.room._id);
+			sendFileRequest(file, roomId, visitor.getToken());
+			parentCall('callback', 'chat-started');
+		});
+	} else {
+		sendFileRequest(file, roomId, visitor.getToken());
+	}
+}
+
+fileUpload = function(file) {
+	Meteor.call('livechat:checkTypeFileUpload', file.file.type, (error, result) => {
+		if (error) {
+			return;
+		}
+
+		if (!result) {
+			swal({
+				title: t('FileUpload_MediaType_NotAccepted'),
+				text: file.type || `*.${ s.strRightBack(file.name, '.') }`,
+				type: 'error',
+				timer: 4000
+			});
+		}	
+		return;
+	});
+	
+	if (file.file.size === 0) {
+		swal({
+			title: t('FileUpload_File_Empty'),
+			type: 'error',
+			timer: 1000,
+			showConfirmButton: false
+		});
+	
+		return;
+	}
+
+	Meteor.call('livechat:checkSizeFileUpload', file.file.size, (error, result) => {
+		if (error) {
+			return;
+		}
+
+		if (!result.result) {
+			swal({
+				title: t('File_exceeds_allowed_size_of_bytes', {size: result.sizeAllowed}),
+				type: 'error',
+				timer: 4000
+			});
+		}
+		return;
+	});
+	
+	return getUploadPreview(file, function(file, preview) {
+		let text = '';
+		if (file.type === 'audio') {
+			text = `\
+			<div class='upload-preview'><audio  style="width: 100%;" controls="controls"><source src="${ preview }" type="audio/wav">Your browser does not support the audio element.</audio></div>`;
+		} else if (file.type === 'video') {
+			text = `\
+			<div class='upload-preview'><video  style="width: 100%;" controls="controls"><source src="${ preview }" type="video/webm">Your browser does not support the video element.</video></div>`;
+		} else if (file.type === 'image') {
+			text = `\
+			<div class='upload-preview'><div class='upload-preview-file' style='background-image: url(${ preview })'></div></div>`;
+		} else {
+			const fileSize = formatBytes(file.file.size);
+			text = `\<div class='upload-preview'><div>${ Handlebars._escape(file.name) } - ${ fileSize }</div></div>`;
+		}
+
+		swal({
+			title: t('Upload_file_question'),
+			text,
+			html: true,
+			showCancelButton: true,
+			cancelButtonText: t('Cancel'),
+			confirmButtonText: t('Send'),
+			closeOnCancel: true,
+			closeOnConfirm: true
+		}, (isConfirm) => {		
+			if (isConfirm) {
```

Comment:
we should try writing more plain functions, it helps readability and maintainability. this one could be rewrite like this:
```javascript
if (!isConfirm) {
	return;
}

const roomId = visitor.getRoom(true);

if (visitor.getId()) {
	return sendFileMessage(file.file, roomId);
}

const guest = {
	token: visitor.getToken()
};

if (Livechat.department) {
	guest.department = Livechat.department;
}

Meteor.call('livechat:registerGuest', guest, (error, result) => {
	if (error) {
		return showError(error.reason);
	}

	visitor.setId(result.userId);
	sendFileMessage(file.file, roomId);
});